src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -71,6 +83,11 @@ public AssemblyService(AssemblyRepository repository, @Qualifier("NCBIDataSource
     }
 
     public void insertAssembly(AssemblyEntity entity) {
+        // Limit cache size to 10 assemblies
+        while (repository.countByIdNotNull() >= 10) {
```

Comment:
While loop instead of an if statement because if two requests reach at once, they both might read cache size < 10 and add an entry leading to cache having more than 10 entries. While loop on next run deletes entities from cache till size < 10. Now of course the same problem can arise if two requests start deleting at the same time but all that will lead to is the cache getting completely emptied.